flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/log/CygnusLogger.java

```diff
@@ -76,7 +112,14 @@ public void warn(String componentName, String message) {
      * @param message
      */
     public void debug(String componentName, String message) {
-        debug("[" + componentName + "] " + message);
+        try {
+            debug("[" + componentName + "] " + message);
+        } catch (Exception e) {
+            if (shutdown) {
+                System.err.println("A problem with the logging system was found... shutting down Cygnus right now!");
```

Comment:
Maybe you should also print the Exception trace/information to provide more information to the administrator reading the mesage?

(BTW, I have copy-paste the same comment 5 times... maybe the code could be refactored to have the print & exit logic in one place :)